### ISSUE TITLE
Feature request:  Password manager integration for account/password filling.

### ISSUE THREAD

**Issue by alanhoyle** [wavebox/waveboxapp#623]
* Which version of Wavebox are you using? [e.g. 3.0.0]

3.12.0

* Which Operating System are you using? [e.g. Windows 10, Mac OSX 10, Ubuntu 14.04]

macOS 10.13.3, Ubuntu 17.10

Wavebox has been really good for integrating many of my services.  One issue I have with it is that I use 1Password to manage my passwords, and I have to copy/paste from the clipboard in order to login.  While I'm aware that there is integration with the macOS Keychain, having multiple separate password stores is annoying at best, and I'd rather have only one place to manage everything.  I primarily use Wavebox on macOS, but I do also have linux and windows workstations.

**Comment by MichaelTunnell** [comment]
Adding support for 1Password would not solve your issue on Ubuntu because *1Password doesn't support Linux*. 1Password requires the desktop app to integrate with browser extensions so it would be impossible to add support for 1Password for usage with Wavebox on Linux.

However, with that said, **Bitwarden would be a great solution** for multiple reasons. Bitwarden works via browser extensions without a requirement of a desktop app. Bitwarden is also open-source and self-hostable. :+1:

**Comment by alanhoyle** [comment]
1Password X supports Chromium cross-platform, including Windows, macOS, and Linux:  https://support.1password.com/getting-started-1password-x/

**Comment by MichaelTunnell** [comment]
I see so 1Password doesn't support Linux but the seemingly hidden (in their site) 1Password X does support it due to being browser exclusive. That is interesting.

I would still recommend Bitwarden be the solution if any solution is chosen because 1Password is $36/year (paid annually) while Bitwarden is Gratis (free) and Open Source.

Bitwarden offers extra features for $10/year if the user wants so it makes more sense to me to be the chosen solution than the locked down premium only 1Password.

**Comment by alanhoyle** [comment]
Hooks for any browser plugin-based password managers would be a nice thing for many people.  1Password also does online vaults as a service and it might be worth considering supporting it as a service in any case.  

AgileBits describes 1PasswordX as "the future of 1P," if their blog posts are to be believed, and they're apparently starting internal testing of Firefox soon.  

I've been using 1Password for just under 10 years now and I'm a subscriber with a family plan, so I have a lot of inertia built up for staying with their product.  Switching password managers requires significant effort just for one person, and that's not something that one takes lightly.  I have difficulty imagining walking my elderly father through a migration.  I just migrated him from KeePass in the past year or so and that took hours.  

If I were starting anew and didn't have others to migrate, Bitwarden might be a good thing to try, but it doesn't seem worth it for me to switch.  I did download the app, and it does seem to work, however I prefer the 1Password "Secret Key"+"yourMasterPassword" model better than just relying on a single master password for use with an online service:  it seems more vulnerable with just one.

**Comment by MichaelTunnell** [comment]
> "Hooks for any browser plugin-based password managers would be a nice thing for many people."

If that is possible to do then that could be great, though I am not aware on the technicals of this.

> "1Password also does online vaults as a service and it might be worth considering supporting it as a service in any case."

It seems that this is a shared vault system and Bitwarden offers that as well. However, 1Password seems to offer file syncing like Dropbox sort of. Bitwarden doesn't do that but there are many great options that offer that, even open source.

> "AgileBits describes 1PasswordX as "the future of 1P," if their blog posts are to be believed, and they're apparently starting internal testing of Firefox-compatible plugin soon."

Overall it sounds like 1Password is very far behind because you have to choose between operating system lockdown or browser lockdown right now. 

> "I've been using 1Password for just under 10 years now and I'm a subscriber with a family plan, so I have a lot of inertia built up for staying with their product."

I understand your hesitation but I'd argue that only further locks you in to what you are already using.

> "Switching password managers requires significant effort just for one person"

Actually, it doesn't require much effort at all with Bitwarden. They have importing tools specifically designed for various other platforms. I moved from LastPass to Bitwarden and it was very easy with practically no effort.

It asked "what do you use now?", it then told me how to export data from LastPass with step by step instructions. I then imported the data into Bitwarden and that was it. Done.

I do not have any experience with 1Password migrating to Bitwarden admittedly but the simplicity of moving from LastPass is likely a sign for other migrations as well.

Documentation from Bitwarden: https://help.bitwarden.com/article/import-from-1password/

>  I have difficulty imagining walking my elderly father through a migration. I just migrated him from KeePass in the past year or so and that took hours.

KeePass is not the most friendly experience in many ways. KeePass itself is combersome. KeePass, KeePass2, KeePassX, KeePassXC are all competitors of the same thing made by different people and those don't even address browsers or mobile. KeePass to anything is a pain.

However, since you already went through that, the next one won't be nearly as bad because KeePass is not in the stack this time.

With Bitwarden you just take a file and import it for them. You could remote into their machine or get their data from the 1Password webclient and import it for them removing all of the headache of walking someone through the process. :)

> "If I were starting anew and didn't have others to migrate, Bitwarden might be a good thing to try, but it doesn't seem worth it for me to switch"

As stated, the migration is not as bad as you might think. I was hesitant at first too.

The family aspect is fine too because Bitwarden offers a family plan just like 1Password so you can move everyone in the same plan if you want. Bitwarden is even cheaper with $1/month for 5 Family Members vs $5/month for 5 Members in 1Password.

> "I did download the app, and it does seem to work, however I prefer the 1Password "Secret Key"+"yourMasterPassword" model better than just relying on a single master password for use with an online service: it seems more vulnerable with just one."

The master password itself should be complicated by default so an extra key isn't really much different if the master password is done properly.

With that said, Bitwarden also goes farther with Two Factor Authentication (2FA) so not only do you have a master password but you also need a physical device of some sort. This is much more security than just a password.

In addition to that, Bitwarden also has taken into consideration the levels of users making something secure yet also easy to use.

Your father might not be comfortable (or interested in) remembering a complicated password, however always being logged in is a security hole. Bitwarden considered this and fixed it. You can set Security PINs per device so he doesn't need to remember the password but rather the 4 Digit PIN instead. This means that even if you do choose to "stay logged in" no one and no other app on the device could access the data without your involvement.

This PIN system is SO NICE that it is one of the key benefits of Bitwarden because it takes something that was cumbersome and annoying for some users and makes it very easy to use while still keeping security since each PIN is specific to each device. (unless of course the user sets the same PIN everywhere)

Overall, I think Bitwarden provides a solid security structure with a well thought out method of convenience. That alone makes it my favorite solution but add onto that the Open Source aspect and the ability to Self-Host the software . . . they have ticked absolutely every box that I require and even some I didn't even think of.

**Comment by alanhoyle** [comment]
Do you have a financial interest in Bitwarden or something? ;-)  I'm sure it's nice.  I'm really not interested in switching my/our password manager.  Besides, WaveBox doesn't seem to have native integration with it, so that's doesn't help what I want:  I'd still need to copy/paste my passwords into WaveBox, and there are tons of iOS apps that have 1Password integration already, so that would be a huge step back for me for all of those.  Maybe Bitwarden will catch up eventually, but for now, it would be less convenient for much of my use.  

1Password doesn't lock me into an OS:  I use it on Mac, Windows, iOS, ChromeOS (both through Android emulation and the 1PX browser extension), and Ubuntu Linux (also through 1PX).  Cost is small-enough in either case and I've already got everything there for me and other members of my family.  

This conversation is like "I'd like Microsoft Teams support" and the response being "You should use Slack."  or "I'd like Hotmail"  with a response being "You should use Roundcube." (I'm aware that the above are supported, which is great, thanks!)  

It's not about whether Bitwarden, 1Password, LastPass, or Dashlane, (or whatever,) is the best password manager.  It's "I'd like password manager support, and I use 1Password for my password manager."

**Comment by chuckreynolds** [comment]
👍 1Password.

**Comment by john1452** [comment]
Any plans for 1 password? I switched from last password 2 days ago and found out wavebox doesn't have support for it :(

**Comment by brennanfee** [comment]
@alanhoyle Do you have a financial interest in Bitwarden or something?

It is open source.  The "interest" benefits us all.  They do have a pay service but only for the syncing features of BitWardne.  For those that don't want to pay them (a very cheap price) or don't want to trust their servers\services the syncing features can be self-hosted.

**Comment by alanhoyle** [comment]
@brennanfee:  While I assume that the above is merely quoting my previous question, I am not a user of Bitwarden, I'm a paying customer of 1Password.  My comment included a ";-)" because I was being encouraged to change to a different product when I have no interest in changing and significant inertia for staying with my current password management setup.  

The existence of an open source alternative does have benefits for the world at large (I've suggested it to friends who've asked me about password managers), and I did test it out to see if it worked.  I do not begrudge anyone for choosing to use Bitwarden.  I use 1Password and I feel that it is/was misplaced to encourage me to use a particular competitor product, open source or not, especially when that product has no better integration within WaveBox than 1Password does.  Open source, in and of itself, is not a sufficient reason to recommend a piece of software.  

I'd still love to have 1Password integration in WaveBox.  I'd rather trust one password manager than having to trust both my manager **and** the WaveBox password storage feature.

**Comment by rmnl** [comment]
+1 for 1Password support

**Comment by jlumbroso** [comment]
@Thomas101 Very happy to (partially?) sponsor 1password integration. Is that a possibility?

Consider that Apple/iOS went from no integration, to integrating all clients through a standard API. Seems like the only fair way to do it for customers, who may be using and loyal to any number of password manager.

**Comment by MichaelTunnell** [comment]
the more and more webapps continue to kick me out of a session the more and more I agree with this feature request regardless of which password manager it is. :D 

though Bitwarden preferably :D

**Comment by DomGrieco** [comment]
+1

I actually went to a different solution for now until Bitwarden support is added to Wavebox.

**Comment by alanhoyle** [comment]
As an aside, some time ago I [asked this question at AgileBits](https://discussions.agilebits.com/discussion/87608/is-is-possible-to-integrate-1password-mini-or-1password-x-with-wavebox-on-macos), and their response was:

>  if Wavebox is Chromium based (I’m not familiar with it enough to say) or supports Chromium-like extensions it may be something we can consider for 1Password X down the road.

**Comment by BertLamb** [comment]
I kinda hope BitWarden support doesn't get added, mainly out of spite. ;)

But I'm also in for a +1 on 1Password X it seems like it should be able to work in the same way that the LastPass support does now.

**Comment by brennanfee** [comment]
@alanhoyle 
> because I was being encouraged to change to a different product when I have no interest in changing

That is perfectly fair.  No worries, we all have the tools we like and are used to.  And I see little reason why Wavebox couldn't support all of them (LastPass, 1Password, and Bitwarden).  Of all browser plugin-like behavior for WaveBox to focus on, the highest priority ought to be these kinds of tools as so many users use them to manage their online accounts.

**Comment by sccolbert** [comment]
Another vote for 1password support.

**Comment by esa1975** [comment]
Another vote from a paid user for Bitwarden support. You already have LastPass so I have to assume it's not a monumental challenge. Great work on the app. This would make it that much better!

**Comment by corpetty** [comment]
+1 for bitwarden support. It's open-source, I'd imagine more alignment with your software development ideals than closed-source counterparts... it has all the relevant integrations for adding as an extension.

**Comment by nolazybits** [comment]
I'll be keen to see keepass (or the http plugin) support.
Using [Keeweb](https://keeweb.info/) and a online storage to store the database file works like a charm and is free.

**Comment by alanhoyle** [comment]
NOTE:  As of 10.x.x, it appears that Wavebox has native support for Chrome Extensions.  

Within a couple minutes of searching the Chrome Web Store, I found extensionsa available for 1Password, Bitwarden, Dashlane, LastPass, KeePass, Roboform.  I've been using the 1Password X extension since I upgraded a few weeks ago, so I would recommend upgrading to Wavebox 10.x and doing a search for a  native extension.
